### ISSUE TITLE
cdc sink error "Write Conflict"

### ISSUE THREAD

**Issue by maxshuang** [pingcap/tiflow#7227]
### What did you do?

From the oncall, 
customer has three cdc nodes, when he starts two cdc nodes at first, the changefeed is normal and the checkpointTs catches up quickly.
After adding a new cdc in the cluster, the checkpointTs of the changefeed lags back.

log:
```
[2022/09/23 07:28:36.822 +00:00] [WARN] [mysql.go:612] ["execute DMLs with error, retry later"] [error="
[CDC:ErrMySQLTxnError]MySQL txn error: Error 9007: Write conflict, txnStartTS=436186150766379010, conflictStartTS=436186150727057454, conflictCommitTS=436186150766379021, 
key={tableID=172, indexID=4, indexValues={1, 85709, 15, 13, }} primary={tableID=161, indexID=1, indexValues={346943, 1850395608943165440, 729430118, 1, 1598468504, }} [try again later]"] 
```
Need more investigation.
https://internal.pingcap.net/jira/browse/ONCALL-5443

### What did you see instead?

--

### Versions of the cluster

Upstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):

```console
(paste TiDB cluster version here)
```

Upstream TiKV version (execute `tikv-server --version`):

```console
(paste TiKV version here)
```

TiCDC version (execute `cdc version`):

```console
v6.1
```

**Comment by nongfushanquan** [comment]
key={tableID=172, indexID=4, indexValues={1, 85709, 15, 13, }} primary={tableID=161, indexID=1, indexValues={346943, 1850395608943165440, 729430118, 1, 1598468504, }} 

since key.tableID is different from primary.tableID, which means a multi-table transaction , some writing-operations by the user in the downstream are expected .

**Comment by dveeden** [comment]
> key={tableID=172, indexID=4, indexValues={1, 85709, 15, 13, }} primary={tableID=161, indexID=1, indexValues={346943, 1850395608943165440, 729430118, 1, 1598468504, }}
> 
> since key.tableID is different from primary.tableID, which means a multi-table transaction , some writing-operations by the user in the downstream are expected .

Are you saying that there are write operations from applications going on on the target cluster? Or is this multiple operations from TiCDC that are conflicting among itself? Would it be possible to print/find the connection ID's from the conflicting transactions?

**Comment by dveeden** [comment]
Would it be possible to:

- show table names (and index/object?)
- show a human readable logical/physical part of the timestamps?
- have a more detailed message if `key.tableID` and `primary.tableID` are not the same?
- have information about the involved threads?

**Comment by nongfushanquan** [comment]
> > key={tableID=172, indexID=4, indexValues={1, 85709, 15, 13, }} primary={tableID=161, indexID=1, indexValues={346943, 1850395608943165440, 729430118, 1, 1598468504, }}
> > since key.tableID is different from primary.tableID, which means a multi-table transaction , some writing-operations by the user in the downstream are expected .
> 
> Are you saying that there are write operations from applications going on on the target cluster? Or is this multiple operations from TiCDC that are conflicting among itself? Would it be possible to print/find the connection ID's from the conflicting transactions?

Yes. The transaction created by the ticdc will change the rows only in one table.
